src/error/parse.rs

```diff
@@ -104,13 +104,34 @@ impl TryFrom<crate::Error> for Parse {
     }
 }
 
-#[cfg(feature = "serde-human-readable")]
+/// Simple wrapper for fmt::Arguments that implements `serde::de::Expected`.
+///
+/// This allows us to use `format_args!()` in a `serde::de::Error`.
+#[cfg(feature = "serde")]
+struct ArgumentsWrapper<'a> {
+    /// Inner wrapped value.
+    args: fmt::Arguments<'a>,
+}
+
+#[cfg(feature = "serde")]
+impl<'a> From<fmt::Arguments<'a>> for ArgumentsWrapper<'a> {
```

Comment:
These lifetimes can be elided.

Reply:
I'm not sure how -- we need to specify that the `ArgumentsWrapper` lives no longer than the `fmt::Arguments`.

I couldn't come up with any combination of elisions that rustc was happy with.

Reply:
No, it doesn't in this case.